release/air_tests/air_benchmarks/workloads/torch_benchmark.py

```diff
@@ -0,0 +1,298 @@
+from torchvision import datasets
+from torchvision.transforms import ToTensor
+
+
+# Define model
+class NeuralNetwork(nn.Module):
+    def __init__(self):
+        super(NeuralNetwork, self).__init__()
+        self.flatten = nn.Flatten()
+        self.linear_relu_stack = nn.Sequential(
+            nn.Linear(28 * 28, 512),
+            nn.ReLU(),
+            nn.Linear(512, 512),
+            nn.ReLU(),
+            nn.Linear(512, 10),
+            nn.ReLU(),
+        )
+
+    def forward(self, x):
+        x = self.flatten(x)
+        logits = self.linear_relu_stack(x)
+        return logits
+
+
+def train_epoch(dataloader, model, loss_fn, optimizer, world_size: int):
+    size = len(dataloader.dataset) // world_size
+    model.train()
+    for batch, (X, y) in enumerate(dataloader):
+        # Compute prediction error
+        pred = model(X)
+        loss = loss_fn(pred, y)
+
+        # Backpropagation
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        if batch % 100 == 0:
+            loss, current = loss.item(), batch * len(X)
+            print(f"loss: {loss:>7f}  [{current:>5d}/{size:>5d}]")
+
+
+def validate_epoch(dataloader, model, loss_fn, world_size: int):
+    size = len(dataloader.dataset) // world_size
+    num_batches = len(dataloader)
+    model.eval()
+    test_loss, correct = 0, 0
+    with torch.no_grad():
+        for X, y in dataloader:
+            pred = model(X)
+            test_loss += loss_fn(pred, y).item()
+            correct += (pred.argmax(1) == y).type(torch.float).sum().item()
+    test_loss /= num_batches
+    correct /= size
+    print(
+        f"Test Error: \n "
+        f"Accuracy: {(100 * correct):>0.1f}%, "
+        f"Avg loss: {test_loss:>8f} \n"
+    )
+    return test_loss
+
+
+def train_func(use_ray: bool, config: Dict):
+    if use_ray:
+        from ray.air import session
+        import ray.train as train
+
+    batch_size = config["batch_size"]
+    lr = config["lr"]
+    epochs = config["epochs"]
+
+    if use_ray:
+        world_size = session.get_world_size()
+    else:
+        world_size = distributed.get_world_size()
+
+    worker_batch_size = batch_size // world_size
+
+    # Load datasets. Use download=False to catch errors in preparation, as the
+    # data should have already been downloaded.
+    training_data = datasets.FashionMNIST(
+        root="/tmp/data_fashion_mnist",
+        train=True,
+        download=False,
+        transform=ToTensor(),
+    )
+
+    test_data = datasets.FashionMNIST(
+        root="/tmp/data_fashion_mnist",
+        train=False,
+        download=False,
+        transform=ToTensor(),
+    )
+
+    # Create data loaders.
+    train_dataloader = DataLoader(training_data, batch_size=worker_batch_size)
+    test_dataloader = DataLoader(test_data, batch_size=worker_batch_size)
+
+    # Create model.
+    model = NeuralNetwork()
+
+    if use_ray:
+        model = train.torch.prepare_model(model)
```

Comment:
Can we also do `prepare_dataloader` for the Ray AIR case, and for the vanilla torch case we have to manually add DistributedSampler and move the tensors to GPU.